core/src/test/scala/neotypes/FutureTransactionUnitSpec.scala

```diff
@@ -0,0 +1,43 @@
+package neotypes
+
+import java.util.concurrent.CompletionStage
+
+import org.mockito.Mockito._
+import org.neo4j.driver.v1.summary.ResultSummary
+import org.scalatest.{Matchers, WordSpec}
+import org.scalatestplus.mockito.MockitoSugar
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+import scala.util.Try
+
+class FutureTransactionUnitSpec extends WordSpec with TransactionUnitSpec[Future] with Matchers with MockitoSugar {
+
```

Comment:
@BalmungSan I thought this test highlights one of the differences between the Future effect type and the others, which is if the neo4j Transaction code throws a runtime exception, the implementation of F.async for future doesn't encapsulate that into a failed future, instead an exception is thrown.

Reply:
That shouldn't happen!
All the effect types should have a roughly equivalent behavior. If one catch some exceptions and the other one doesn't, we need to refactor the code!